experanto/utils.py

```diff
@@ -165,4 +169,267 @@ def __iter__(self):
             yield k, next(loader)
 
     def __len__(self):
-        return len(self.loaders) * self.min_batches
\ No newline at end of file
+        return len(self.loaders) * self.min_batches
+
+def replace_nans_with_neighbors(data):
+    """
+    Replace NaN values in a tensor with the previous value if available,
+    otherwise with the next valid value.
+
+    Args:
+        data (torch.Tensor): 2D tensor of shape (N, 2) where NaNs may exist.
+
+    Returns:
+        torch.Tensor: Tensor with NaNs replaced.
+    """
+    data = data.clone()  # Create a copy to avoid modifying the original tensor
+    for i in range(data.size(0)):  # Iterate over rows
+        if torch.isnan(data[i]).any():
+            if i > 0 and not torch.isnan(data[i - 1]).any():  # Use previous value
+                data[i] = data[i - 1]
+            elif i < data.size(0) - 1 and not torch.isnan(data[i + 1]).any():  # Use next value
+                data[i] = data[i + 1]
+            else:
+                # If no previous or next valid value, replace with zeros or a default value
+                data[i] = torch.zeros_like(data[i])
+    return data
+
+def get_validation_split(n_images, train_frac, seed):
+    """
+    Splits the total number of images into train and test set.
+    This ensures that in every session, the same train and validation images are being used.
+
+    Args:
+        n_images: Total number of images. These will be plit into train and validation set
+        train_frac: fraction of images used for the training set
+        seed: random seed
+
+    Returns: Two arrays, containing image IDs of the whole imageset, split into train and validation
+
+    """
+    if seed:
+        np.random.seed(seed)
+    train_idx, val_idx = np.split(
+        np.random.permutation(int(n_images)), [int(n_images * train_frac)]
+    )
+    assert not np.any(
+        np.isin(train_idx, val_idx)
+    ), "train_set and val_set are overlapping sets"
+
+    return train_idx, val_idx
+
+class GazeBasedCrop(torch.nn.Module):
```

Comment:
@PavithraElumalai I might move this class to a new .py file like `augmentations.py` (we would probably have more augmentations later)